packages/babel-plugin-proposal-decorators/test/fixtures/2023-01-misc/all-decorators/output.js

```diff
@@ -2,52 +2,50 @@ var _initProto, _initStatic, _initClass, _init_a, _init_d, _init_e, _call_f, _ca
 const dec = () => {};
 let _Class;
 new class extends babelHelpers.identity {
-  static {
-    class Class {
-      static {
-        ({
-          e: [_init_m, _call_o, _call_p, _call_q, _init_r, _get_r, _set_r, _init_d, _call_f, _call_g, _call_g2, _init_h, _get_h, _set_h, _init_i, _init_n, _init_a, _init_e, _initProto, _initStatic],
-          c: [_Class, _initClass]
-        } = babelHelpers.applyDecs2301(this, [[dec, 7, "j"], [dec, 8, "k"], [dec, 9, "l"], [dec, 6, "m"], [dec, 7, "o", function () {}], [dec, 8, "p", function () {}], [dec, 9, "q", function (v) {}], [dec, 6, "r", o => o.#D, (o, v) => o.#D = v], [dec, 2, "b"], [dec, 3, "c"], [dec, 4, "c"], [dec, 1, "d"], [dec, 2, "f", function () {}], [dec, 3, "g", function () {}], [dec, 4, "g", function (v) {}], [dec, 1, "h", o => o.#B, (o, v) => o.#B = v], [dec, 5, "i"], [dec, 5, "n", o => o.#n, (o, v) => o.#n = v], [dec, 0, "a"], [dec, 0, "e", o => o.#e, (o, v) => o.#e = v]], [dec], _ => #e in _));
-        _initStatic(this);
-      }
-      #f = _call_f;
-      a = (_initProto(this), _init_a(this));
-      b() {}
-      get c() {}
-      set c(v) {}
-      #A = _init_d(this);
-      get d() {
-        return this.#A;
-      }
-      set d(v) {
-        this.#A = v;
-      }
-      #e = _init_e(this);
-      get #g() {
-        return _call_g(this);
-      }
-      set #g(v) {
-        _call_g2(this, v);
-      }
-      #B = _init_h(this);
-      set #h(v) {
-        _set_h(this, v);
-      }
-      get #h() {
-        return _get_h(this);
-      }
-      static j() {}
-      static get k() {}
-      static set l(v) {}
-      static get m() {
-        return this.#C;
-      }
-      static set m(v) {
-        this.#C = v;
-      }
+  static [class Class {
+    static {
+      ({
+        e: [_init_m, _call_o, _call_p, _call_q, _init_r, _get_r, _set_r, _init_d, _call_f, _call_g, _call_g2, _init_h, _get_h, _set_h, _init_i, _init_n, _init_a, _init_e, _initProto, _initStatic],
+        c: [_Class, _initClass]
+      } = babelHelpers.applyDecs2301(this, [[dec, 7, "j"], [dec, 8, "k"], [dec, 9, "l"], [dec, 6, "m"], [dec, 7, "o", function () {}], [dec, 8, "p", function () {}], [dec, 9, "q", function (v) {}], [dec, 6, "r", o => o.#D, (o, v) => o.#D = v], [dec, 2, "b"], [dec, 3, "c"], [dec, 4, "c"], [dec, 1, "d"], [dec, 2, "f", function () {}], [dec, 3, "g", function () {}], [dec, 4, "g", function (v) {}], [dec, 1, "h", o => o.#B, (o, v) => o.#B = v], [dec, 5, "i"], [dec, 5, "n", o => o.#n, (o, v) => o.#n = v], [dec, 0, "a"], [dec, 0, "e", o => o.#e, (o, v) => o.#e = v]], [dec], _ => #e in _));
+      _initStatic(this);
     }
```

Comment:
Maybe we can optimize this output in the future?

Reply:
Sure. Though the output here is only one trailing `;` more than what we currently have. The output change here is introduced in https://github.com/babel/babel/pull/16325/commits/2d34eeb216b49f918ed0f34c279b9c65da9d4e52, which changes the output `static { class { ... } }` to `static [ class { ... } ];`.

Reply:
I'm not sure if I can merge them into one class. :)
The changes here look good to me, the properties are kept in the outer class which greatly improves readability.

Reply:
> merge them into one class

The non static elements are split because the static fields have to be installed on the replaced class. If we can merge them without too much effort, https://github.com/babel/babel/issues/16176 will be automatically resolved.